loki/transformations/hoist_variables.py

```diff
@@ -139,9 +141,13 @@ def transform_subroutine(self, routine, **kwargs):
         if role != 'driver':
             variables = self.find_variables(routine)
             item.trafo_data[self._key]["to_hoist"] = variables
+            dims = flatten([getattr(v, 'shape', []) for v in variables])
+            item.trafo_data[self._key]["imported_sizes"] = [(d.type.module, d) for d in dims
+                                                            if str(d) in routine.import_map]
```

Comment:
This rebuilds the import_map in every iteration. Better to cache it before the list comprehension. 

```suggestion
            import_map = routine.import_map
            item.trafo_data[self._key]["imported_sizes"] = [(d.type.module, d) for d in dims
                                                            if d.name in import_map]
```